docs/docs/92-development/05-architecture.md

```diff
@@ -6,4 +6,41 @@
 
 ## System architecture
 
+### main package hirarchie
+
+| package    | meaning                                                      | import
+|------------|--------------------------------------------------------------|----------
+| `cmd/**`   | parse commanline args & environment to stat server/cli/agent | all other
+| `agent/**` | code only agent (remote worker) will need                    | `pipeline`, `shared`
+| `cli/**`   | code only cli tool does need                                 | `pipeline`, `shared`, `woodpecker-go`
+| `server/**`| code only server will need                                   | `pipeline`, `shared`
+| `shared/**`| code shared for all three main tools (go help utils)         | only std and external libs
+| `woodpecker-go/**` | go client for server rest api                        | std
+
+### Server
+
+| package            | meaning                                         | import
+|--------------------|-------------------------------------------------|----------
+| `server/api/**`    | handle web requests from `server/router`        | `pipeline`, `server/(badges\|ccmenue\|logging\|model\|pubsub\|queue\|remote\|shared\|store)`, `shared`, (TODO: mv `server/router/middleware/session`)
```

Comment:
as we can see we should move more stuff out of it

either create a new package if we can "discover" a new scope or distribute it to other packages ...

Reply:
cc @anbraten